src/lambda/gc01_check_monitoring_and_logging/app.py

```diff
@@ -74,26 +74,26 @@ def evaluate_parameters(rule_parameters):
     """
     return rule_parameters
 
-def list_cloudtrails():
+def list_cloud_trails() -> list[dict]:
     """Fetches a list of all trails in the account"""
     try:
-        response = AWS_CLOUDTRAIL_CLIENT.list_trails()
-        trails = response.get("Trails")
-        next_token = response.get("NextToken")
-        while next_token != None:
-            response = AWS_CLOUDTRAIL_CLIENT.list_trails()
-            trails = trails + response.get("Trails")
+        trails = []
+        next_token = None
+        while True:
+            response = AWS_CLOUDTRAIL_CLIENT.list_trails() if not next_token else AWS_CLOUDTRAIL_CLIENT.list_trails(NextToken=next_token)
+            trails = trails + response.get("Trails", [])
             next_token = response.get("NextToken")
+            if not next_token:
+                break
```

Comment:
Updated to use the loop style, but also needed to pass in the next token to the `list_trails` call when it is available.